pkg/autoscaling/calculation.go

```diff
@@ -281,21 +282,37 @@ func getScaledGroupsByComponent(rc *cluster.RaftCluster, component ComponentType
 
 func getScaledTiKVGroups(informer core.StoreSetInformer, healthyInstances []instance) []*Plan {
 	planMap := make(map[string]map[string]struct{}, len(healthyInstances))
+	resourceTypeMap := make(map[string]string)
 	for _, instance := range healthyInstances {
 		store := informer.GetStore(instance.id)
 		if store == nil {
 			log.Warn("inconsistency between health instances and store status, exit auto-scaling calculation",
 				zap.Uint64("store-id", instance.id))
 			return nil
 		}
-		v := store.GetLabelValue(groupLabelKey)
-		buildPlanMap(planMap, v, instance.address)
+
+		groupName := store.GetLabelValue(groupLabelKey)
+		if !isAutoScaledGroup(groupName) {
+			continue
+		}
+
+		buildPlanMap(planMap, groupName, instance.address)
+
+		if _, ok := resourceTypeMap[groupName]; !ok {
+			resourceType := store.GetLabelValue(resourceTypeLabelKey)
+			if resourceType == "" {
+				log.Warn("store is in auto-scaled group but has no resource type label", zap.Uint64("store-id", instance.id))
```

Comment:
If the precondition is not meet (as we didn't find resourceType label here), I think we should throw an error here and directly quit the calculation.